src/module/core-modules/napa/timer-wrap.cpp

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#include <vector>
+#include <memory>
+#include <iostream>
+
+#include "timer-wrap.h"
+
+#include <napa/zone.h>
+#include <zone/worker.h>
+#include <zone/scheduler.h>
+#include <zone/worker-context.h>
+#include <zone/async-context.h>
+#include <zone/callback-task.h>
+
+using namespace napa::module;
+
+using napa::zone::WorkerId;
+using napa::zone::WorkerContext;
+using napa::zone::WorkerContextItem;
+using napa::zone::NapaZone;
+using napa::zone::SchedulePhase;
+
+using v8::Array;
+using v8::Boolean;
+using v8::Context;
+using v8::Function;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::Global;
+using v8::HandleScope;
+using v8::Isolate;
+using v8::Local;
+using v8::Number;
+using v8::Object;
+using v8::Persistent;
+using v8::String;
+using v8::Value;
+
+NAPA_DEFINE_PERSISTENT_CONSTRUCTOR(TimerWrap);
+
+void TimerWrap::Init() {
+    auto isolate = Isolate::GetCurrent();
+    auto constructorTemplate = FunctionTemplate::New(isolate, DefaultConstructorCallback<TimerWrap>);
+    constructorTemplate->SetClassName(v8_helpers::MakeV8String(isolate, exportName));
+    constructorTemplate->InstanceTemplate()->SetInternalFieldCount(1);
+
+    NAPA_SET_PERSISTENT_CONSTRUCTOR(exportName, constructorTemplate->GetFunction());
+}
+
+Local<Object> TimerWrap::NewInstance(std::shared_ptr<napa::zone::Timer> timer) {
+    auto object = napa::module::NewInstance<TimerWrap>().ToLocalChecked();
+    auto wrap = NAPA_OBJECTWRAP::Unwrap<TimerWrap>(object);
+    wrap->_timer = std::move(timer);
+    return object;
+}
+
+void TimerWrap::Reset() {
+    _timer.reset();
+}
+
+napa::zone::Timer& TimerWrap::Get() {
+    return *_timer;
+}
+
+static void dummyWeakCallback(const v8::WeakCallbackInfo<int>& data) {
+}
+
+std::shared_ptr<napa::zone::CallbackTask> buildTimeoutTask(
```

Comment:
Let's define class CallbackTask inside this file and keep it local. There is no reason to introduce a public null task implementation.

Reply:
Hide CallbackTask inside the only related cpp file.